### PR TITLE
DriverState update

### PR DESCRIPTION
**Pull request by KmakD**
### Description
Due to changes in panther_msgs DriverState (https://github.com/husarion/panther_msgs/pull/79). This makes stack work with new msg, tests are passing. This does not aim to refactor anything with lynx in mind, just update the stack so everything still works while having universal DriverState msg. 

[Summary of the changes]

### Modifications

-

### CODE REVIEW

File: panther_battery/README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Publishes battery state read from ADC unit for Panther version 1.2 and above, or
 #### Subscribes
 
 - `hardware/io_state` [*panther_msgs/IOState*]: Current state of IO.
-- `hardware/motor_controllers_state` [*panther_msgs/DriverState*]: Current motor controllers' state and error flags. Subscribed if using Roboteq motor controllers data.
+- `hardware/robot_driver_state` [*panther_msgs/DriverState*]: Current motor controllers' state and error flags. Subscribed if using Roboteq motor controllers data.
```
Comment (pawelirh) [inline panther_battery/README.md]: Incorrect type

File: panther_hardware_interfaces/src/panther_system/panther_system_ros_interface.cpp [hunk 2]
```diff
@@ -290,4 +301,32 @@ rclcpp::CallbackGroup::SharedPtr PantherSystemRosInterface::GetOrCreateNodeCallb
...
+  front_driver_state.name = DriverStateNamedMsg::NAME_FRONT;
+  rear_driver_state.name = DriverStateNamedMsg::NAME_REAR;
+
+  auto & driver_state = realtime_driver_state_publisher_->msg_;
+  driver_state.drivers_states.push_back(front_driver_state);
+  driver_state.drivers_states.push_back(rear_driver_state);
+}
+
+DriverStateNamedMsg & PantherSystemRosInterface::GetDriverStateByName(
+  RobotDriverStateMsg & robot_driver_state, const std::string & name)
```
Comment (pawelirh) [inline panther_hardware_interfaces/src/panther_system/panther_system_ros_interface.cpp:317]:
```suggestion
  const RobotDriverStateMsg & robot_driver_state, const std::string & name)
```

File: panther_hardware_interfaces/README.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ Plugin responsible for communicating with engine controllers via the CAN bus, pr
 - `diagnostics` [*diagnostic_msgs/DiagnosticArray*]: Panther system diagnostic messages.
 - `hardware/e_stop` [*std_msgs/Bool*]: Current E-stop state.
 - `hardware/io_state` [*panther_msgs/IOState*]: Current IO state.
-- `hardware/motor_controllers_state` [*panther_msgs/DriverState*]: Current motor controllers' state and error flags.
+- `hardware/robot_driver_state` [*panther_msgs/DriverState*]: Current motor controllers' state and error flags.
```
Comment (pawelirh) [inline panther_hardware_interfaces/README.md]: Wrong type

File: panther_manager/test/test_safety_manager_node.cpp [hunk 4]
```diff
@@ -194,12 +194,14 @@ TEST_F(TestSafetyManagerNode, DriverStateCBBlackboardUpdate)
 {
   const float expected_temp = 21.0;
 
-  auto driver_state_msg = panther_msgs::msg::DriverState();
-  driver_state_msg.front.temperature = expected_temp;
-  driver_state_msg.rear.temperature = expected_temp;
+  panther_msgs::msg::DriverStateNamed motor_controller;
```
Comment (pawelirh) [inline panther_manager/test/test_safety_manager_node.cpp]: Why `motor_controller`?

File: panther_manager/README.md [hunk 5]
```diff
@@ -58,7 +58,7 @@ Node responsible for managing safety features, and software shutdown of componen
 - `battery/battery_status` [*sensor_msgs/BatteryState*]: State of the internal Battery.
 - `hardware/e_stop` [*std_msgs/Bool*]: State of emergency stop.
 - `hardware/io_state` [*panther_msgs/IOState*]: State of IO pins.
-- `hardware/motor_controllers_state` [*panther_msgs/DriverState*]: State of motor controllers.
+- `hardware/robot_driver_state` [*panther_msgs/DriverState*]: State of motor controllers.
```
Comment (pawelirh) [inline panther_manager/README.md]: type missing

**Review by pawelirh — CHANGES_REQUESTED** [review]

File: panther_hardware_interfaces/src/panther_system/panther_system_ros_interface.cpp (hunk 2 above)
Comment (KmakD) [reply panther_hardware_interfaces/src/panther_system/panther_system_ros_interface.cpp:317]: This method returns a non-const reference, so this param can't be read-only. It won't compile

**Review by pawelirh — APPROVED** [review]

**Comment by pawelirh** [comment]
Once the PR on `panther_msgs` is merged, we can update its version in the `.respos` file. Afterwards the PR is to be merged 👍🏼

**Pull request merged**